src/btree/bt_page.c

```diff
@@ -329,8 +329,8 @@ __wt_page_inmem_prepare(WT_SESSION_IMPL *session, WT_REF *ref)
  *     Build in-memory page information.
  */
 int
-__wt_page_inmem(WT_SESSION_IMPL *session, WT_REF *ref, const void *image, uint32_t flags,
-  WT_PAGE **pagep, bool *preparedp)
+__wt_page_inmem(WT_SESSION_IMPL *session, WT_REF *ref, const void *image, size_t image_alloc_size,
+  uint32_t flags, WT_PAGE **pagep, bool *preparedp)
```

Comment:
I've been thinking about this change, because I'm concerned that it isn't obvious when we decide which size to track. Is it possible to make the change without this associated change to the input parameters here? i.e: call `wt_page_inmem` figure out if it should use the image allocation size or the disk allocation size?

Or, maybe there could be a variable in the WT_PAGE structure that always has the tracked size, and this function just uses that?

Reply:
Well, the key point is to consistently track the malloced size. In this function, the page is created within the function itself, meaning it hasn't been created yet. 

Determining the `image_alloc_size` within this function is a challenge if it is not directly passed as a parameter. However, I will spend some time thinking if there is a way.

Reply:
It appears challenging to incorporate this parameter into WT_REF (as we aim to avoid increasing the size of this struct). Another concern lies with the void* image, (which is persisted on disk, and modifying it may lead to compatibility issues). Additionally, the WT_PAGE cannot be modified since it has not been created yet.

One potential solution is to pass the WT_ITEM variable (instead of the const void*) and extract the image and size from it. However, this approach could pose problems if the function is called from `__split_multi_inmem`, as it does not have a WT_ITEM struct and instead uses WT_MULTI.

Reply:
@agorrod after re-reviewing this, I think it is best to pass the parameter (explicitly) itself.